### PR TITLE
Self-identification: tweak wording & add other section

### PR DESCRIPTION
**Pull request by qaisjp**
## What this PR does / Why we need it
- Tweaks wording from asking about gender to a general "What do you identify as?"
- Allow for "Other" option

## Some questions
- [x] I have read the contributing guidelines
- [x] I abide by this repository Code of Conduct
- [x] I understand that my PR won't be merged until Travis gives a "green light"

## Additional Notes (optional)
Do you want to add anything else? We :heart: to hear your opinions!

@vaidap — your feedback on this would be amazing!

Also, generally: I am not sure what happens to this new column when you export hacker data!

### CODE REVIEW

File: applications/migrations/0014_gender_other.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.4 on 2019-02-11 22:15
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
```
Comment (qaisjp) [inline applications/migrations/0014_gender_other.py:10]: Should `dependencies` be empty? I think this is autopopulated by the migration generator.

Comment (casassg) [reply applications/migrations/0014_gender_other.py:10]: No, it needs to have the previous one in order to work. Used to keep in DB what migrations have been applied.

File: applications/models.py [hunk 2]
```diff
@@ -114,6 +116,8 @@ class Application(models.Model):
     # ABOUT YOU
     # Population analysis, optional
     gender = models.CharField(max_length=20, choices=GENDERS, default=NO_ANSWER)
+    other_gender = models.CharField(max_length=20, blank=True, null=True)
```
Comment (casassg) [inline applications/models.py]: Is 20 chars going to be enough?

**Review by casassg — APPROVED** [review]
lgtm overall. I think it may be cool to add the data from MyMLH if available as well. I think they have a field from their platform that is also other gender.

**Comment by vaidap** [comment]
thanks for doing this @qaisjp! ❤️ 
a few comments:
- it sounds a bit weird to remove "gender" from the question-- I feel like "What gender do you identify as?" sounds better.
- I'd replace "Other" with "Prefer to self-describe", as per this example: https://twitter.com/underskinnyhrt/status/1086275803917111297
- As for whether 20 chars is a good max length, i'd increase it to 30-40 if that's not an issue on the backend, as I can think of a few examples that would go beyond 20 chars.

If you're interested, the general advice I follow is the one summarised in @kitation 's slides: https://slidr.io/kitation/asking-about-gender-merseycode#22 and the one right after for an example.

**Comment by qaisjp** [comment]
Thanks for the review Vaida!

@casassg, any other thoughts?

**Review by casassg — APPROVED** [review]
Nothing else to add! Thanks for working on this! 🙌

**Pull request merged**